lib/reversetunnel/agentpool.go

```diff
@@ -675,7 +675,8 @@ func (c *agentPoolRuntimeConfig) updateRemote(ctx context.Context, addr *utils.N
 		if ok := errors.As(err, &tls.RecordHeaderError{}); !ok {
 			return trace.Wrap(err)
 		}
-	} else {
+		// Only use the ping results if they weren't from a minimal api handler on the reverse tunnel port.
+	} else if addr.Addr != ping.Proxy.SSH.TunnelListenAddr || ping.Proxy.SSH.TunnelListenAddr == ping.Proxy.SSH.WebListenAddr {
 		tlsRoutingEnabled = ping.Proxy.TLSRoutingEnabled
 	}
```

Comment:
Suggestion: simplify flow / incorporate info about minimal handler.

```suggestion
	}

	// Only use the ping results if they weren't from a minimal handler.
	// The minimal API handler only exists when the proxy and reverse tunnel are
	// listening on separate ports, so it will never do TLS routing.
	isMinimalHandler := /* write condition here */
	if !isMinimalHandler {
		tlsRoutingEnabled = ping.Proxy.TLSRoutingEnabled
	}
```